Tests/RelayerTests/AutomaticSocketConnectionHandlerTests.swift

```diff
@@ -55,12 +55,15 @@ final class AutomaticSocketConnectionHandlerTests: XCTestCase {
         XCTAssertFalse(webSocketSession.isConnected)
     }
 
-    func testReconnectOnDisconnectForeground() {
+    func testReconnectOnDisconnectForeground() async {
         appStateObserver.currentState = .foreground
         XCTAssertTrue(webSocketSession.isConnected)
         webSocketSession.disconnect()
         sut.handleDisconnection()
-        XCTAssertTrue(webSocketSession.isConnected)
+        Task { @MainActor in
+            // Simulate time for reconnect
+            XCTAssertTrue(webSocketSession.isConnected)
+        }
```

Comment:
I think this test will always pass even if you comment out `sut.handleDisconnection()`
what about adding an expectation?

Reply:
Good catch! I was thinking `await Task.sleep` but what I have is a separate task not awaited for. I've added an `expectation` like you suggested and asserting / fulfilling within the async execution.